src/test/mir-opt/match_false_edges.rs

```diff
@@ -0,0 +1,170 @@
+    let _ = match Some(1) {
+        Some(_w) if guard() => 1,
+        _x => 2,
+        Some(y) if guard2(y) => 3,
+        _z => 4,
+    };
+}
+
+// END RUST SOURCE
+//
+// START rustc.node17.NLL.before.mir
+//  bb0: {
+//      ...
+//      _2 = std::option::Option<i32>::Some(const 42i32,);
+//      _5 = discriminant(_2);
+//      switchInt(_5) -> [0isize: bb6, otherwise: bb7];
+//  }
+//  bb1: { // arm1
+//      StorageLive(_7);
+//      _7 = _3;
+//      _1 = Add(const 1i32, _7);
+//      ...
+//      goto -> bb11;
+//  }
+//  bb2: { // binding1 guard
+//      StorageLive(_3);
+//      _3 = ((_2 as Some).0: i32);
+//      StorageLive(_6);
+//      _6 = const guard() -> bb8;
+//  }
+//  bb3: { // binding2 & arm2
+//      StorageLive(_4);
+//      _4 = ((_2 as Some).0: i32);
+//      StorageLive(_8);
+//      _8 = _4;
+//      _1 = Add(const 2i32, _8);
+//      StorageDead(_8);
+//      goto -> bb11;
+//  }
+//  bb4: { // binding3(empty) arm3
+//      _1 = const 3i32;
+//      goto -> bb11;
+//  }
+//  bb5: {
+//      unreachable;
+//  }
+//  bb6: {
+//      falseEdges -> [real: bb4, imaginary: bb5]; // from before_binding3 to unreachable
+//  }
+//  bb7: {
+//      falseEdges -> [real: bb2, imaginary: bb3]; // from before_binding1 to binding2
+//  }
+//  bb8: {
+//      switchInt(_6) -> [0u8: bb9, otherwise: bb1]; // end of guard
+//  }
+//  bb9: {
+//      falseEdges -> [real: bb10, imaginary: bb3]; // after_guard to binding2
+//  }
+//  bb10: {
+//      falseEdges -> [real: bb3, imaginary: bb4]; // from before_binding2 to binding3
+//  }
+//  bb11: {
+//      ...
+//      return;
+//  }
+//
+//
+// END rustc.node17.NLL.before.mir
+//
+// START rustc.node40.NLL.before.mir
+// bb0: {
+//     ...
+//     _2 = std::option::Option<i32>::Some(const 1i32,);
+//     _7 = discriminant(_2);
+//     switchInt(_7) -> [1isize: bb8, otherwise: bb11];
+// }
+// bb1: { // arm1
+//      _1 = const 1i32;
+//      goto -> bb15;
+// }
+// bb2: { // arm3
+//     _1 = const 3i32;
+//      goto -> bb15;
+// }
+// bb3: { // binding1: Some(w) if guard() =>
+//     StorageLive(_3);
+//     _3 = ((_2 as Some).0: i32);
+//     StorageLive(_8);
+//     _8 = const guard() -> bb9;
+// }
+// bb4: { // binding2 & arm2
+//     StorageLive(_4);
+//     _4 = _2;
+//     _1 = const 2i32;
+//     goto -> bb15;
+// }
+// bb5: { // binding3: Some(y) if guard2(y) =>
+//     StorageLive(_5);
+//     _5 = ((_2 as Some).0: i32);
+//     StorageLive(_10);
+//     StorageLive(_11);
+//     _11 = _5;
+//    _10 = const guard2(_11) -> bb12;
+// }
+// bb6: { // binding4 & arm4
+//     StorageLive(_6);
+//     _6 = _2;
+//     _1 = const 4i32;
+//     goto -> bb15;
+// }
+// bb7: {
+//     unreachable;
+// }
+// bb8: {
+//     falseEdges -> [real: bb3, imaginary: bb4]; // from before_binding1 to binding2
+// }
+// bb9: {
+//     switchInt(_8) -> [0u8: bb10, otherwise: bb1]; // end of gurard
+// }
+// bb10: {
+//     falseEdges -> [real: bb11, imaginary: bb4]; // after guard to binding2
```

Comment:
This should go to `before_binding2` rather than `binding2`, because otherwise there won't be a path from the guard